src/live_restore/live_restore_private.h

```diff
@@ -95,6 +95,31 @@ struct __wti_live_restore_fs_layer {
     WTI_LIVE_RESTORE_FS_LAYER_TYPE which;
 };
 
+/*
+ * Live restore states. As live restore progresses we will transition through each of these states
+ * one by one. Live restore transitions through each state in the order they are listed below.
+ */
+typedef enum {
+    /*
+     * This is not a valid state. We return it when there is no state file on disk and therefore
+     * we're not in live restore yet.
+     */
+    WTI_LIVE_RESTORE_STATE_NONE,
+    /*
+     * The background migration state is where the majority is where the majority of work takes
+     * place. Users can perform reads/writes while we copy backing data to the destination in the
+     * background.
+     */
+    WTI_LIVE_RESTORE_STATE_BACKGROUND_MIGRATION,
+    /* We've completed background migration and are now cleaning up any live restore metadata. */
+    WTI_LIVE_RESTORE_STATE_CLEAN_UP,
+    /* We've completed the live restore. */
+    WTI_LIVE_RESTORE_STATE_COMPLETE
+} WTI_LIVE_RESTORE_STATE;
+
+#define WTI_LIVE_RESTORE_MIGRATION_COMPLETE(state) \
```

Comment:
Is it possible to wind in the `get_state` call here? Practically every usage only uses the returned state var once, in this check.

Reply:
Done in https://github.com/wiredtiger/wiredtiger/pull/11590/commits/d3f3f9327ea22d654b1efc0ff9c301d964d91f6b